### PR TITLE
boost 1.71.0

### PR DESCRIPTION
**Pull request by henryiii**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by henryiii** [comment]
Didn't realize that was going to go all the way and make a full PR without asking. Anyway, I wanted to start the ball rolling on this. I think it might also fix #41515 without having to use an old Boost.

**Comment by henryiii** [comment]
So far, we have:

- [ ] avro-cpp
- [ ] cclive
- [ ] cucumber-cpp
- [ ] ledger
- [ ] lgogdownloader
- [ ] libbitcoin-client
- [ ] libbitcoin-explorer
- [ ] libbitcoin-protocol
- [ ] libbitcoin-server
- [ ] libpulsar
- [ ] mkvtoolnix 
- [ ] monero
- [ ] ncmpcpp
- [ ] source-highlight
- [ ] widelands

Several of them seem to just need to be rebuilt (looking for missing symbol, usually from the regex library).

**Comment by henryiii** [comment]
Bumped revisions on failing formula, bumped the other boost packages to 1.71, and dropped the CMake config suppression, since the comment said to drop it in the next version - we will see if it's fixed.

**Comment by henryiii** [comment]
Should we drop or rename `boost-python` to `boost-python2`, and make `boost-python3` into `boost-python`?

**Comment by scpeters** [comment]
we had to modify 48 files for boost 1.70 ( https://github.com/Homebrew/homebrew-core/pull/38890/ ), and I see some of them in the failed parts of this current build

File: Formula/boost-build.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class BoostBuild < Formula
   desc "C++ build system"
   homepage "https://www.boost.org/build/"
-  url "https://github.com/boostorg/build/archive/boost-1.70.0.tar.gz"
-  sha256 "6630adb18e9fdddf354ce16ee7c358fa79aa0ae264da3b5604cbed6769ce84e5"
+  url "https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.bz2"
```
Comment (scpeters) [inline Formula/boost-build.rb]: I think we need to keep using github's boostorg/build tarballs:

* https://github.com/boostorg/build/archive/boost-1.71.0.tar.gz

Comment (henryiii) [reply Formula/boost-build.rb]: That's odd, I don't think I changed this.

Comment (henryiii) [reply Formula/boost-build.rb]: Ah, I copied and pasted between the bump formula and didn't notice that this URL was different.

**Comment by scpeters** [comment]
The builds are ongoing, but I'm peeking at a few early failures.

There's a [Formula reports different SHA256](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/47556/version=mojave/testReport/junit/brew-test-bot/mojave/fetch___build_from_source_libbitcoin_protocol/) for libbitcoin-protocol, and its GitHub release page indicates that it was released 2 days ago, even though we merged it into homebrew 25 days ago, so I created an upstream issue to ask if they modified their tag: https://github.com/libbitcoin/libbitcoin-protocol/issues/210

* https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/47556/version=mojave/testReport/junit/brew-test-bot/mojave/fetch___build_from_source_libbitcoin_protocol/

There's a [compilation failure in libpulsar](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/47556/version=mojave/testReport/junit/brew-test-bot/mojave/install___build_bottle_libpulsar/) that has been fixed upstream in https://github.com/apache/pulsar/pull/4972, though it is caused by opportunistic linkage to snappy, so it would be best to explicitly disable snappy support or add it as an explicit dependency. I looked at the [CMakeLists.txt](https://github.com/apache/pulsar/blob/master/pulsar-client-cpp/CMakeLists.txt#L188-L193), but I'm not sure if `-DHAS_SNAPPY=0` would work.

**Comment by fxcoudert** [comment]
General comment: the pull request should have one commit per formula (whether it's a version update, revision bump, or any other modification).

Then, the `boost-python@1.70` alias needs to be renamed `boost-python@1.71`:
```
boost-python:
  * Formula has other versions so create a versioned alias:
      cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases
      ln -s ../Formula/boost-python.rb boost-python@1.71
  * Formula has invalid versioned aliases:
      boost-python@1.70
```

Other than that, it's actually going pretty well. We need to have another PR with updates to the `libbitcoin-protocol` and `libbitcoin-client` checksums, which we will merge before this one, so we can have a clean round of testing.

**Comment by henryiii** [comment]
> I'm not sure if -DHAS_SNAPPY=0 would work.

I'm 80% sure that local variables trump cached ones, so this would not work.

**Comment by henryiii** [comment]
@fxcoudert Should I group related formula? Like boost-*, libbitcoin-*, or is it always one commit per file?

**Comment by fxcoudert** [comment]
> Seems to be one commit per file.

Yes. The only things we group are changes to aliases (creation and renaming). The rest is always one commit per formula. Make its easier to track down.

**Comment by fxcoudert** [comment]
> I'm ready to push once the other 1-2 PRs get merged

You can, but you'll have to rebase on top of master first, those PRs created conflicts.

File: Formula/boost-build.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class BoostBuild < Formula
   desc "C++ build system"
   homepage "https://www.boost.org/build/"
   url "https://github.com/boostorg/build/archive/boost-1.70.0.tar.gz"
```
Comment (scpeters) [inline Formula/boost-build.rb]: I think we lost the url change here

Comment (henryiii) [reply Formula/boost-build.rb]: Wow, that's odd. Not sure how I managed that.

**Comment by henryiii** [comment]
Failures are:

* brew-test-bot.mojave.test libbitcoin-protocol
* brew-test-bot.mojave.test libbitcoin-client
* brew-test-bot.mojave.test libbitcoin-explorer
* brew-test-bot.mojave.test libbitcoin-server

Missing symbol for regex (why are these not rebuilding?).

* brew-test-bot.mojave.fetch avro-cpp --build-bottle
* brew-test-bot.mojave.install --build-bottle avro-cpp

Cannot download avro-cpp.

* brew-test-bot.mojave.test cclive

```
Test::Unit::AssertionFailedError: <0> expected but was <1>.
```

* brew-test-bot.mojave.install --build-bottle ledger

```
CMake Error at /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:117 (find_package):
  Could not find a package configuration file provided by "boost_python"
  (requested version 1.71.0) with any of the following names:

    boost_pythonConfig.cmake
    boost_python-config.cmake

  Add the installation prefix of "boost_python" to CMAKE_PREFIX_PATH or set
  "boost_python_DIR" to a directory containing one of the above files.  If
  "boost_python" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:182 (boost_find_component)
  /usr/local/Cellar/cmake/3.15.2/share/cmake/Modules/FindBoost.cmake:422 (find_package)
  CMakeLists.txt:88 (find_package)
```

**Comment by henryiii** [comment]
I expect we can set `boost_python_DIR` to fix ledger. Don't know why the libbitcoin recipes are getting an old symbol. I assume Avro-cpp will fix itself next time, which leaves cclive; I'm not sure what's failing there.

**Comment by henryiii** [comment]
Correction: It seems like the avro-cpp link does not work anymore since a new version was released (1.9.1), so the old version now needs an archive link. Should I just update this to the new version in this PR? Or switch to the archive link? Does apache.org provide a nicer link that doesn't change when a new version is released?

**Comment by scpeters** [comment]
it's possible that some of the failures might be related to the `--no-cmake-config` change?

**Comment by henryiii** [comment]
The `boost_python_DIR` for ledger I believe is directly related to that change; I don't think the libbitcoin errors are related to it. cclive is not really reporting why it is failing, so I don't know what the problem is there.

**Comment by henryiii** [comment]
Since I'm modifying the commits and force pushing, the changes in the current build are: libbitcoin (base) incremented, Avro-cpp version bumped, and the ledger config directory set (hopefully not missing a subdirectory).

**Comment by scpeters** [comment]
if they cmake change is preventing the bitcoin packages from finding the regex library components, then it could be related; I'm not sure how the boost cmake config file works

**Comment by scpeters** [comment]
but it's more likely that incrementing `libbitcoin` is the solution and that the cmake config changes are unrelated, just noticed that you made that change

**Comment by henryiii** [comment]
Down to two failures, cclive and ledger.

I should be able to fix ledger for now by setting `Boost_NO_BOOST_CMAKE`, which will keep the find package from using the external CMake files; we probably should check to make sure the Boost-Python config files are being produced and installed.

CCLive isn't producing anything helpful as to why it's dying. It was last updated in 2013, so no recent info about it's support for Boost, etc. Any ideas?

**Comment by henryiii** [comment]
PS: In the last 30 days, cclive seems to have been downloaded 6 times, if I read the analytics correclty.

**Comment by jonchang** [comment]
Testing locally...

```console
$ cclive https://youtu.be/VaVZL7F6vqU
Checking ... ...... ...... libquvi: error: ...cripts/share/libquvi-scripts/lua/website/youtube.lua:152: no match: media title
```

Looks like it's a problem in `libquvi`, which also hasn't been updated in a while. It seems unrelated to the `boost` update, but rather that Youtube has changed its APIs and the parsers haven't been updated.

I'd be OK with deleting `cclive`. Alternatively we could change the test to `cclive -S https://youtu.be/VaVZL7F6vqU` which just emits media formats and seems to work ok.

**Comment by fxcoudert** [comment]
@henryiii can you comment on the last force push? I'm not sure I see what you changed.

**Comment by henryiii** [comment]
Yes, I somehow changed the sha265 and comment for libpulsar to move to 2.4.1, but didn't change the version number - that's why it was the (one) failure. The force push actually changed `2.4.0` to `2.4.1`.

The fact that the URL for Apache downloads becomes invalidated as soon as a new release is made is *very* irritating - it basically makes any PR with apache formula a time bomb; if a new version is released it no longer builds. Is there any way we can avoid using those URLs, or provide a backup URL with the archive version? Maybe use the GitHub release for them instead? (Assuming they are all on GitHub too, at least libpulsar seemed to be).

**Comment by henryiii** [comment]
If #43904 is merged, then libpulstar's change needs to be just changed to a revision bump (and I think the comment change about 2.4.1+ supporting snappy is still a good idea).

**Comment by henryiii** [comment]
The same issue happened with avro-cpp #43814, by the way.

**Comment by henryiii** [comment]
Tests pass! :) Let me know if you'd rather merge #43904, then I'll have to rebase and change the version bump to a revision bump, and hope no other Apache formula update in the mean time.

**Comment by scpeters** [comment]
Well done! I think we can do libpulsar from here. I'll leave it to @fxcoudert so this doesn't interfere with #43860

**Comment by fxcoudert** [comment]
Congratulations and many many thanks @henryiii!

**Pull request closed without merging**